scripts/prepare_heat_demand.py

```diff
@@ -0,0 +1,394 @@
+
+    Parameters
+    ----------
+    demands : pd.DataFrame
+         DataFrame that contains yearly demands
+    value : float
+         Value of the yearly demand
+    consumer : str
+         Name of the consumer (eg.: ghd, efh, mfh)
+    carrier : str
+         Name of carrier (eg.: heat_central, heat_decentral)
+
+    Returns
+    -------
+    demands : pd.DataFrame
+         Updated DataFrame that contains yearly demands
+    """
+    if consumer + "_" + carrier in demands.columns:
+        demands[consumer + "_" + carrier].values[0] = np.add(
+            demands[consumer + "_" + carrier].values[0], value
+        )
+    else:
+        demands[consumer + "_" + carrier] = [value]
+    return demands
+
+
+def get_heat_demand(path, scenario, carrier):
+    """
+    This function returns ghd and hh demands together with their unit of a given region and a
+    given scenario
+
+    Parameters
+    ----------
+    path : str
+        input path
+    scenario : str
+        scenario e.g. "base"
+    carrier : str
+         Name of carrier (eg.: heat_central, heat_decentral)
+
+    Returns
+    -------
+    demand : DataFrame
+        Dataframe with total yearly demand of central and decentral heat per consumer
+        (eg.: ghd, efh, mfh)
+    demand_unit : str
+        unit of total demands
+
+    """
+    # Read state heat demands of ghd and hh sectors
+    sc = dp.load_b3_scalars(path)  # TODO: Add ', sep=";"' if from schema
+    consumers = ["ghd", "efh", "mfh"]
+    demands = pd.DataFrame()
+    newline = "\n"
+
+    sc_filtered = dp.filter_df(sc, "tech", "demand")
+    sc_filtered = dp.filter_df(sc_filtered, "carrier", carrier)
+    sc_filtered = dp.filter_df(sc_filtered, "region", region)
+    sc_filtered = dp.filter_df(sc_filtered, "scenario", scenario)
+
+    if not (sc_filtered["var_unit"].values[0] == sc_filtered["var_unit"].values).all():
+        raise ValueError(
+            f"Unit mismatch in scalar data of heat demands. "
+            f"Please make sure units match in {path}."
+        )
+
+    demand_unit = list(set(sc_filtered["var_unit"]))
+
+    for consumer in consumers:
+        sc_filtered_consumer = sc_filtered[sc_filtered["name"].str.contains(consumer)]
+
+        if len(sc_filtered_consumer) > 1:
+            print(
+                f"User warning: There is duplicate demand of carrier '{carrier}', consumer "
+                f"'{consumer}', region '{region}' and scenario '{scenario}' in {path}."
+                + newline
+                + f"The demand is going to be summed up. "
+                "Otherwise you have to rerun the calculation and provide only one demand of the "
+                "same carrier, consumer, region and scenario."
+            )
+
+        for demand_value in sc_filtered_consumer["var_value"].values:
+            check_central_decentral(demands, demand_value, consumer, carrier)
+
+    return demands, demand_unit
+
+
+def calculate_heat_load():
```

Comment:
Please pass all variables necessary for this calculation explicitly.

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/76/commits/a22b2ed677dd336b9f9c741508ef149a357fae1c.